### PR TITLE
Add leaf node

### LINKED ISSUE #228: Implement Tree leaf node, with extra information
From previous discussions, we decided to have a collapsed item (users can expand it if necessary) with extra information.

While no replacement for the table, it will provide extra information that is essential for managing workflows.

### PR DESCRIPTION
**Pull request by kinow**
Closes #228

### CODE REVIEW

**Comment by kinow** [comment]
And ready for review! :tada:

**Comment by hjoliver** [comment]
(Conflicts - probably due to just merged progress function PR?)

**Comment by hjoliver** [comment]
One thing @kinow (not a blocker for this PR though - it's also unsorted on master): cycle points need to be date-time ordered:
Screenshot from 2019-09-17 17-32-34

**Comment by kinow** [comment]
@hjoliver if you could re-review, please.

- Fixed conflicts (not easy to fix when you have mocked data, but hopefully didn't mess it up)
- See last commit https://github.com/cylc/cylc-ui/pull/233/commits/23b2ff6a09f1548968e87510c09d67f332b9f76b, it sorts `taskProxies` by `cyclePoint`

Perhaps @dwsutherland could be the second reviewer if he has time? To review this one, just need to run Cylc UI + Hub, and check if the GraphQL query is OK (for both the extra data for leaf nodes, and now for sorting `taskProxies`).

I'll do a quick manual testing now to confirm progress and leaf node are working as expected :+1:

**Comment by kinow** [comment]
Progress and leaf node both working for me:

image

GraphQL sort done with `taskProxies(sort: { keys: ["cyclePoint"] }) {` (following example that @dwsutherland kindly provided before for `jobs` :tada: )

File: src/components/cylc/TreeItem.vue [hunk 1]
```diff
@@ -106,7 +128,38 @@ export default {
...
+        {
+          title: 'start time',
+          property: 'startedTime'
+        },
+        {
+          title: 'finish time',
+          property: 'finishedTime'
+        },
+        {
+          title: 'last message',
```
[the remark this answers is not in the document]
Comment (kinow) [reply src/components/cylc/TreeItem.vue]: I used latest initially.. the only issue I had was with small devices. Using `xs1` + `latest message`, the string is too long and breaks the UI.

Using `xs2` + `latest message` works, but other titles stay too much to the left :disappointed_relieved: 

Should we just use latest here and worry about the responsiveness later?

[the remark this answers is not in the document]
Comment (wxtim) [reply src/components/cylc/TreeItem.vue]: Think that makes sense.

[the remark this answers is not in the document]
Comment (matthewrmshin) [reply src/components/cylc/TreeItem.vue]: An innocent suggestion is often the cause of breakage!

Up to you to pick:
1. Ignore my suggestion?
2. Ignore the breakage for responsiveness?
3. Maybe just *latest* without *message*?

**Comment by kinow** [comment]
Changed text to `latest message` (thanks @matthewrmshin !) and created https://github.com/cylc/cylc-ui/issues/236 to discuss the issue :+1:

**Pull request merged**